src/main/java/org/candlepin/service/model/Event.java

```diff
@@ -0,0 +1,48 @@
+/*
+ * Copyright (c) 2009 - 2024 Red Hat, Inc.
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+package org.candlepin.service.model;
+
+/**
+ * The Event interface represents a Candlepin event.
+ */
+public interface Event extends ServiceAdapterModel {
```

Comment:
Shouldn't we also be creating the implementation of the CloudCheckInEvent class for this interface as part of this task? (In the next task ConsumerManager.updateLastCheckIn would be creating an instance of it, so it could also be created as part of that one?)

Reply:
Oh good catch! I thought that the implementation was to be done in the ConsumerManager task (CANDLEPIN-899), but it is not listed on that task. It does make sense to add the implementation to this PR. I will get that added

Reply:
Added the `CloudCheckInEvent` class